src/test/java/org/broadinstitute/hellbender/utils/recalibration/covariates/ContextCovariateUnitTest.java

```diff
@@ -46,52 +51,97 @@ public void testSimpleContexts() {
         }
     }
 
-    public static void verifyCovariateArray(int[][] values, int contextSize, GATKRead read, Covariate contextCovariate, final byte lowQualTail) {
+    private static void verifyCovariateArray(final int[][] values, final int contextSize, final GATKRead read, final Covariate contextCovariate, final byte lowQualTail) {
         for (int i = 0; i < values.length; i++) {
             Assert.assertEquals(contextCovariate.formatKey(values[i][0]), expectedContext(read, i, contextSize, lowQualTail), "offset " + i);
         }
     }
 
-    @DataProvider(name="strandedBytes")
+    @DataProvider
+    Iterator<Object[]> AnnoyingReads() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+
+        final String[] bases = new String[] {
+//           A string of all IUPAC bases
+                "NBTRMLNACMGRSVTWYHKDBNABTRMLNBTRML",
```

Comment:
This string has some ACGT bases in it, despite being described as "A string of all IUPAC bases". I think the "all IUPAC" case is definitely worth covering here...